### PR TITLE
[NetManager] Integrate auth service tenancy

### PR DESCRIPTION
**Pull request by ja-odur**
#### Summary of Changes (What does this PR do?)
- Enables auth-service tenancy support

#### Is this change ready to hit production in its current state?
- [ ] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [ ] I consider this code done
- [ ] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

Good to have (up to reviewer's discretion):
- [ ] I've tested this on staging
- [ ] Unit test coverage of changes

#### How should this be manually tested?
* Pull and checkout to this [branch]() locally
* Install node requirements `npm install`
* Ensure the local auth-service with data-segregation is running. Check this [PR](https://github.com/airqo-platform/AirQo-api/pull/121) for more info
* You start the application in one of the two ways:
  1. `npm run dev-mac` (on mac) or `npm run dev-pc` (on windows platform)
  2. `npm run prod-mac` (on mac) or `npm run prod-pc` (on windows platform) and manually change the auth-service url configs to point to the locally running service. The benefit of this approach is you are able to load other data required by the app (such as analytics).
* On the login page, a new field `organization` is required. Using the current credentials and organization, login should be successful.

#### What are the relevant tickets?
- [PLAT-164](https://airqoteam.atlassian.net/browse/PLAT-164)

#### Screenshots (optional)
* The login page screenshot
Screenshot 2020-09-23 at 15 48 51

### CODE REVIEW

File: netmanager/src/redux/Join/actions.js [hunk 1]
```diff
@@ -351,7 +342,7 @@ export const registrationSuccess = data => {
 /************************* Login a new User  *********************************/
 export const loginUser = userData => dispatch => {
   console.log('the login URL ' + constants.LOGIN_USER_URI);
-  const tenant = 'kcca'
+  const tenant = userData.organization || "kcca"
```
Comment (Baalmart) [inline netmanager/src/redux/Join/actions.js:345]: Does this mean that if the organisation does not exist, use "kcca"?

On a related note, just entered some random org name and was unable to proceed, could we add some alert to inform user about the issue ("your organisation does not exist on the platform").
On that note, I have made a new commit to the [corresponding backend PR](https://github.com/airqo-platform/AirQo-api/pull/121) (please PULL) to send an appropriate error response (see below) which we can use accordingly.
```
{
    "success": false,
    "message": "organization does not exist"
}
```

Comment (Baalmart) [reply netmanager/src/redux/Join/actions.js:345]: I do not have clear recommendations about how we can handle this scenario but feel free to explore and choose best approach. :)

**Review by Baalmart — CHANGES_REQUESTED** [review]
Thanks @ja-odur 
Works as expected, just one small UX issue we shall need to address.

File: netmanager/src/redux/Join/actions.js (hunk 1 above)
Comment (ja-odur) [reply netmanager/src/redux/Join/actions.js:345]: > I do not have clear recommendations about how we can handle this scenario but feel free to explore and choose best approach. :)

Noted. Thank you. I will fix this.

Comment (ja-odur) [reply netmanager/src/redux/Join/actions.js:345]: > Does this mean that if the organisation does not exist, use "kcca"?
> 
> On a related note, just entered some random org name and was unable to proceed, could we add some alert to inform user about the issue ("your organisation does not exist on the platform").
> On that note, I have made a new commit to the [corresponding backend PR](https://github.com/airqo-platform/AirQo-api/pull/121) (please PULL) to send an appropriate error response (see below) which we can use accordingly.
> 
> ```
> {
>     "success": false,
>     "message": "organization does not exist"
> }
> ```

I will have to remove the or part. You cannot reach this point without providing an org. will fix and work on displaying any errors back to the user

Comment (ja-odur) [reply netmanager/src/redux/Join/actions.js]: > Does this mean that if the organisation does not exist, use "kcca"?
> 
> On a related note, just entered some random org name and was unable to proceed, could we add some alert to inform user about the issue ("your organisation does not exist on the platform").
> On that note, I have made a new commit to the [corresponding backend PR](https://github.com/airqo-platform/AirQo-api/pull/121) (please PULL) to send an appropriate error response (see below) which we can use accordingly.
> 
> ```
> {
>     "success": false,
>     "message": "organization does not exist"
> }
> ```

@Baalmart This has been addressed

**Review by Baalmart — CHANGES_REQUESTED** [review]
Things are taking shape, thanks @ja-odur 
I ran in production and manually changed the auth service production configs to point to a locally run auth-service [using this PR(branch)](https://github.com/airqo-platform/AirQo-api/pull/121). Just a few observations I want to leave here: 

- [ ] The [data segregation PR](https://github.com/airqo-platform/AirQo-api/pull/121) was updated this morning to protect all the endpoints for auth-service. I hope that change has not affected performance here at all. Asking because I am not able to checkout the `users` and `candidates` pages.

- [ ] User Experience: The user is still left wondering whenever wrong ORG is entered on the Login page. 

- [ ] The pages for `users `and `candidates` are currently experiencing a runtime error. More details in the screenshots.

let me share some screenshots below:
image

image

image

image

image

image

**Comment by ja-odur** [comment]
> Things are taking shape, thanks @ja-odur
> I ran in production and manually changed the auth service production configs to point to a locally run auth-service [using this PR(branch)](https://github.com/airqo-platform/AirQo-api/pull/121). Just a few observations I want to leave here:
> 
> * [ ]  The [data segregation PR](https://github.com/airqo-platform/AirQo-api/pull/121) was updated this morning to protect all the endpoints for auth-service. I hope that change has not affected performance here at all. Asking because I am not able to checkout the `users` and `candidates` pages.
> * [ ]  User Experience: The user is still left wondering whenever wrong ORG is entered on the Login page.
> * [ ]  The pages for `users `and `candidates` are currently experiencing a runtime error. More details in the screenshots.
> 
> let me share some screenshots below:
> image
> 
> image
> 
> image
> 
> image
> 
> image
> 
> image

I have failed to replicate this. Tested this with the new updated data segregation PR and works fine. maybe a sync to get more info would be helpful.

**Review by Baalmart — APPROVED** [review]
works as expected

**Pull request merged**